File: src/main/scala/io/izzel/taboolib/util/item/inventory/stored/ActionQuickTake.java
package io.izzel.taboolib.util.item.inventory.stored;

import io.izzel.taboolib.util.item.ItemStacker;
import io.izzel.taboolib.util.item.inventory.ClickEvent;
import org.bukkit.inventory.ItemStack;

/**
 * @Author sky
 * @Since 2019-12-03 19:20
 */
public class ActionQuickTake extends Action {

    @Override
    public ItemStack getCurrent(ClickEvent e) {
        return e.getClicker().getItemOnCursor();
    }

    @Override
    public void setCurrent(ClickEvent e, ItemStack item) {
        if (item != null) {
            ItemStacker.moveItemFromChest(item, e.getClicker());
        }
        e.getClicker().setItemOnCursor(null);
    }

    @Override
    public int getCurrentSlot(ClickEvent e) {
        return e.getRawSlot();
    }
}

File: module-ui/src/main/kotlin/taboolib/module/ui/type/ActionQuickTake.kt
package taboolib.module.ui.type

import org.bukkit.inventory.ItemStack
import taboolib.common.Isolated
import taboolib.module.ui.ClickEvent
import taboolib.module.ui.ItemStacker
import taboolib.platform.util.isNotAir

@Isolated
class ActionQuickTake : Action() {

    override fun getCursor(e: ClickEvent): ItemStack {
        return e.clicker.itemOnCursor
    }

    override fun setCursor(e: ClickEvent, item: ItemStack?) {
        if (item.isNotAir()) {
            ItemStacker.MINECRAFT.moveItemFromChest(item, e.clicker)
        }
        e.clicker.setItemOnCursor(null)
    }

    override fun getCurrentSlot(e: ClickEvent): Int {
        return e.rawSlot
    }
}